stdlib/public/core/FloatingPointToString.swift

```diff
@@ -0,0 +1,2413 @@
+/// try to compensate for the following:
+/// * The output ergonomics described above do take some time.
+///   It would be faster to always emit the form "123456e-78"
+//    (See `finishFormatting()`)
+/// * The implementations in published papers generally include
+///   large tables with every power of 10 computed out.  We've
+///   factored these tables down to conserve code size, which
+///   requires some additional work to reconstruct the needed power
+///   of 10. (See the `intervalContainingPowerOf10_*` functions)
+
+///
+/// This Swift implementation was ported from an earlier C version;
+/// the output is exactly the same in all cases.
+/// A few notes on the Swift transcription:
+/// * We use MutableSpan<UTF8.CodeUnit> and MutableRawSpan to
+///   identify blocks of working memory.
+/// * We use unsafe/unchecked operations extensively, supported by
+///   several years of analysis and testing of the original C
+///   implementation to ensure that no unsafety actually occurs.  For
+///   Float32, that testing was exhaustive -- we verified all 4
+///   billion possible Float32 values.
+/// * The Swift code uses an idiom of building up to 8 digit characters
+///   in a UInt64 and then writing the whole block to memory.
+/// * The Swift version is slightly faster than the C version;
+///   mostly thanks to various minor algorithmic tweaks that were
+///   found during the translation process.
+/// * Most of this file is annotated for SwiftStdlib 6.2
+///   because it relies on UInt128, MutableSpan, and InlineArray.
+///
+// ----------------------------------------------------------------------------
+
+
+// ================================================================
+//
+// Float16
+//
+// ================================================================
+
+// Float16 is not currently supported on Intel macOS.
+// (This will change once there's a fully-stable Float16
+// ABI on that platform.)
+#if !((os(macOS) || targetEnvironment(macCatalyst)) && arch(x86_64))
+
+// Support Legacy ABI on top of new implementation
+@available(SwiftStdlib 6.2, *)
+@_silgen_name("swift_float16ToString")
+public func _float16ToStringImpl(
+  _ textBuffer: UnsafeMutablePointer<UTF8.CodeUnit>,
+  _ bufferLength: UInt,
+  _ value: Float16,
+  _ debug: Bool) -> UInt64 {
+    // Code below works with raw memory.
+    var buffer = unsafe MutableSpan<UTF8.CodeUnit>(_unchecked: textBuffer,
+                                                   count: Int(bufferLength))
+    let textRange = Float16ToASCII(value: value, buffer: &buffer)
+    let textLength = textRange.upperBound - textRange.lowerBound
+
+    // Move the text to the start of the buffer
+    if textRange.lowerBound != 0 {
+        unsafe _memmove(dest: textBuffer,
+                        src: textBuffer + textRange.lowerBound,
+                        size: UInt(truncatingIfNeeded: textLength))
+    }
+    return UInt64(truncatingIfNeeded: textLength)
+}
```

Comment:
This file's formatting is a bit out of sync with [standard library style](https://github.com/swiftlang/swift/blob/main/docs/StandardLibraryProgrammersManual.md#coding-style) – the main fixes I see are:
- Please use 2-space indents instead of 4
- In function calls that have to wrap their parameters, add a break before the first parameter, instead of after:
  ```swift
  var buffer = unsafe MutableSpan<UTF8.CodeUnit>(
    _unchecked: textBuffer,
    count: Int(bufferLength))
   ```
- For functions like this one, where the parameter list wraps, add a break before the closing parenthesis:
  ```swift
  public func _float16ToStringImpl(
    _ textBuffer: UnsafeMutablePointer<UTF8.CodeUnit>,
    _ bufferLength: UInt,
    _ value: Float16,
    _ debug: Bool
  ) -> UInt64 {
  ```